File: web/src/app/company/Team.ts
import { Employee } from "./Employee";
import { Company } from "./company";

export class Team {
    id: number;
    name: string;
    company: Company;
    employees: Employee[]

    public constructor(
        id: number,
        name: string,
        company: Company,
        employees: Employee[]
    ) {
        this.id = id
        this.name = name
        this.company = company
        this.employees = employees
    }
}

export class TeamShort {
  public constructor(
    public id: number,
    public name: string,
  ) {}
}

export class TeamsListResponse {
    success: boolean;
    data: Team[];

    public constructor(
        success: boolean,
        data: Team[]
    ) {
        this.success = success;
        this.data = data
    }
}

export class TeamCreateRequest {
    name: string;
    employees: number[];

    public constructor(
        name: string,
        employees: number[]
    ) {
        this.name = name;
        this.employees = employees
    }
}

export class TeamCreateResponse {
    success: boolean;
    data: Team;

    public constructor(
        success: boolean,
        data: Team
    ) {
        this.success = success;
        this.data = data
    }
}

File: mobile/app/src/main/java/com/example/abc_jobs_alpaca/model/models/Team.kt
package com.example.abc_jobs_alpaca.model.models

import com.google.gson.Gson
import org.json.JSONObject

data class Team(
        val id: Int,
        val name: String,
        val company: CompanyDetails,
        val employees: List<Employee>
)

data class TeamsResponse(val success: Boolean, val data: List<Team>?)

fun deserializeTeams(response: JSONObject): TeamsResponse {
    val gson = Gson()
    return gson.fromJson(response.toString(), TeamsResponse::class.java)
}

data class TeamItem(val id: Int, val name: String)
